docs/azure/sdk/authentication/create-token-credentials-from-configuration.md

```diff
@@ -0,0 +1,135 @@
+
+## Support for Azure credentials through configuration
+
+The [`Microsoft.Extensions.Azure`](https://www.nuget.org/packages/Microsoft.Extensions.Azure) library can automatically provide Azure service clients with a `TokenCredential` class by searching _appsettings.json_ or other configuration files for credential values using the default `IConfiguration` service for .NET. This approach allows developers to explicitly set credential values across different environments through configuration rather than through app code directly.
+
+The following credential types are supported via configuration:
+
+* [ClientCertificateCredential](#create-a-clientcertificatecredential-type)
+* [ClientSecretCredential](#create-a-clientsecretcredential-type)
+* [DefaultAzureCredential](#create-a-defaultazurecredential-type)
+* [ManagedIdentityCredential](#create-a-managedidentitycredential-type)
+* [WorkloadIdentityCredential](#create-a-workloadidentitycredential-type)
+
+## Configure Azure credentials
+
+Azure service clients registered with the `AddAzureClients` method are automatically configured with an instance of `DefaultAzureCredential`. You can override the global `DefaultAzureCredential` using credential values from configuration files when registering a client:
+
+```csharp
+builder.Services.AddAzureClients(clientBuilder =>
+{
+    // Register BlobServiceClient using credentials from appsettings.json
+    clientBuilder.AddBlobServiceClient(builder.Configuration.GetSection("Storage"));
+
+    // Register ServiceBusClient using the fallback DefaultAzureCredential credentials
+    clientBuilder.AddServiceBusClientWithNamespace(
+        "<your_namespace>.servicebus.windows.net");
+});
+```
+
+The associated _appsettings.json_ file:
+
+```json
+"Storage": {
+    "serviceUri": "<service_uri>",
+    "credential": "managedidentity",
+    "clientid":  "<clientId>"
+}
+```
+
+### Create a `ManagedIdentityCredential` type
+
+You can create both user-assigned and system-assigned managed identities using configuration values. Add the following configuration values to your _appsettings.json_ file to create an <xref:Azure.Identity.ManagedIdentityCredential?displayProperty=fullName>.
+
+#### User-assigned identities
+
+1. Authenticate using a client ID:
+
+    ```json
+    {
+        "credential": "managedidentity",
+        "clientId":  "<clientId>"
+    }
+    ```
+
+1. Authenticate using a resource ID:
+
+    ```json
+    {
+        "credential": "managedidentity",
+        "resourceId":  "<managedIdentityResourceId>"
+    }
+    ```
+
+    The resource ID takes the form `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}`.
+
+#### System-assigned identities
+
+1. Authenticate using a client ID:
+
+    ```json
+    {
+        "credential": "managedidentity",
+        "clientid":  "<clientId>"
```

Comment:
This should be omitted to select a system-assigned MI. Related to the comment above, we should validate that this works (I think it would).

Reply:
I'm not sure I follow - that would mean the only value would be "credential": "managedIdetity". Based on this code below it seems like you have to provide a resourceId or clientId to create a managedIdentityCredential:

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs#L108-L120

Reply:
I'm also not sure how to test this scenario since the fallback is DefaultAzureCredential, so even if specifically a system assigned or user assigned through configuration doesn't get picked up properly, it would still work using the fallback DefaultAzureCredential?

Reply:
To clarify - set  `"credential": "managedidentity"` and omit `clientid`

Reply:
Do you have any suggestions on how to test this? If I deploy an app with a registered service client to Azure I'm not sure how to verify whether its creating a system assigned identity using the config file or if its just falling back to DefaultAzureCredential since all of this happens internally.

Reply:
Nothing about the config will create a system assigned identity - it's just whether or not the config specifying `"credential": "managedidentity"` without a clientId will succeed. If it does, this should mean that we created a DAC with no options, which defaults to system-assigned identity.

Reply:
Sorry if I'm not understanding this right - if you specify "credential": "managedidentity" without a clientId I believe that would end up at the line linked below and pass in a null value for clientId, which would create a ManagedIdentityCredential, not DefaultAzureCredential - is this what you'd expect?

Reply:
>  is this what you'd expect?

Yes. Do you expect something different?

Reply:
No, just making sure I understood your previous message - thanks.